File: src/lib/protocol/nip36.ts
import type { NostrEvent, NostrTag } from './event';

export function contentWarningReason(event: NostrEvent): string | null {
  const tag = event.tags.find((item) => item[0] === 'content-warning');
  if (!tag) return null;
  return tag[1]?.trim() || '';
}

export function hasContentWarning(event: NostrEvent): boolean {
  return contentWarningReason(event) !== null;
}

export function contentWarningTag(reason = ''): NostrTag {
  const trimmed = reason.trim();
  return trimmed ? ['content-warning', trimmed] : ['content-warning'];
}

File: crates/lkjstr-protocol/src/nip36.rs
use crate::{NostrEvent, NostrTag};

pub fn content_warning_reason(event: &NostrEvent) -> Option<String> {
    event
        .tags
        .iter()
        .find(|tag| tag.first().is_some_and(|name| name == "content-warning"))
        .map(|tag| tag.get(1).map_or("", String::as_str).trim().to_owned())
}

pub fn has_content_warning(event: &NostrEvent) -> bool {
    content_warning_reason(event).is_some()
}

pub fn content_warning_tag(reason: &str) -> NostrTag {
    let trimmed = reason.trim();
    if trimmed.is_empty() {
        vec!["content-warning".to_owned()]
    } else {
        vec!["content-warning".to_owned(), trimmed.to_owned()]
    }
}
